platforms/core-configuration/declarative-dsl-provider/src/integTest/groovy/org/gradle/internal/declarativedsl/project/DeclarativeDSLCustomDependenciesExtensionsSpec.groovy

```diff
@@ -383,6 +465,30 @@ class DeclarativeDSLCustomDependenciesExtensionsSpec extends AbstractIntegration
         """
     }
 
+    private String defineDeclarativeDSLProducerBuildScript() {
+        return """
+            plugins {
+                id("com.example.restricted")
+            }
+
+            library {}
+        """
+    }
+
+    private String defineDeclarativeDSLConsumerBuildScript() {
+        return """
+            plugins {
+                id("com.example.restricted")
+            }
+
+            library {
+                dependencies {
+                    api(project(":producer"))
```

Comment:
Given that we currently also have `Project#project`, should we also somehow check that it is `Dependencies#project` that is invoked here? Or would this not even work before the change?

Reply:
Yep, see previous conversation.